src/zeep/xsd/visitor.py

```diff
@@ -802,10 +802,11 @@ def visit_union(self, node, parent):
         """
         # TODO
         members = node.get('memberTypes')
+        default_namespace = node.nsmap.get(None)
         types = []
         if members:
             for member in members.split():
-                qname = as_qname(member, node.nsmap, self.document._target_namespace)
+                qname = as_qname(member, node.nsmap, default_namespace or self.document._target_namespace)
```

Comment:
Take this example:

```
<?xml version="1.0"?>
<schema
    xmlns="http://www.w3.org/2001/XMLSchema"
    xmlns:tns="http://tests.python-zeep.org/"
    targetNamespace="http://tests.python-zeep.org/"
    elementFormDefault="qualified">
  <element name="item" type="tns:Date"/>
  <simpleType name="Date">
    <union memberTypes="date tns:MMYY"/>
  </simpleType>
  <simpleType name="MMYY">
    <restriction base="string" />
  </simpleType>
</schema>
```

Looking up type `date` for the `Date` simpleType will fail because it will try to look up the type in the targetNamespace. My understanding is that it *should* be looked up in the default namespace because it doesn't have a prefix, and the default namespace exists. 

Only when *defining* a type it should be defined in the targetNamespace, in this case its about referring to a type.

I encoutered this problem while trying to use the zeep library with the Kadaster, Dutch government service.